pkg/cmdlinefilter/cmdlinefilter.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2019 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package cmdlinefilter provides filters to modify the bootloader kernel
+// parameters before booting.
+package cmdlinefilter
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"unicode"
+
+	"github.com/u-root/u-root/pkg/cmdline"
+)
+
+// Remove filter the given cmdline by removing fields.
+func Remove(cl string, fields []string) string {
+	var newCl []string
+
+	// kernel variables must allow '-' and '_' to be equivalent in variable
```

Comment:
But just in the names, right? What about - and _ in variable values? I think you may only do these in variable names, but not values.

Reply:
See canonicalKey below

Reply:
Yes maybe "fields" is the wrong name, those are kernel variable name

Reply:
Hopefully the new doc is more clear